Content.Server/CriminalRecords/Systems/GeneralCriminalRecordConsoleSystem.cs

```diff
@@ -0,0 +1,289 @@
+            {
+                { SecurityStatus.Detained, Loc.GetString("general-criminal-record-console-detained-with-reason", ("name", msg.Name), ("reason", msg.Reason)!, ("goodguyname", Name(msg.Session.AttachedEntity.Value))) },
+                { SecurityStatus.None, Loc.GetString("general-criminal-record-console-undetained-with-reason", ("name", msg.Name), ("reason", msg.Reason)!, ("goodguyname", Name(msg.Session.AttachedEntity.Value))) }
+            };
+
+            _radioSystem.SendRadioMessage(uid, messages[secInfo.Status], _prototypeManager.Index<RadioChannelPrototype>("Security"), uid);
+        }
+
+        else if (msg.Reason == string.Empty && msg.Name != null && msg.Session.AttachedEntity != null && secInfo != null)
+        {
+            secInfo.Status = secInfo.Status == SecurityStatus.Detained ? SecurityStatus.None : SecurityStatus.Detained;
+            secInfo.Reason = msg.Reason;
+
+            var messages = new Dictionary<SecurityStatus, string>()
+            {
+                { SecurityStatus.Detained, Loc.GetString("general-criminal-record-console-detained-without-reason", ("name", msg.Name), ("goodguyname", Name(msg.Session.AttachedEntity.Value))) },
+                { SecurityStatus.None, Loc.GetString("general-criminal-record-console-undetained-without-reason", ("name", msg.Name), ("goodguyname", Name(msg.Session.AttachedEntity.Value))) }
+            };
+
+            _radioSystem.SendRadioMessage(uid, messages[secInfo.Status], _prototypeManager.Index<RadioChannelPrototype>("Security"), uid);
+        }
+
+        var station = _stationSystem.GetOwningStation(msg.Session.AttachedEntity!.Value);
+
+        TryComp<StationRecordsComponent>(station, out var stationRecordsComponent);
+
+        _stationRecordsSystem.TryGetRecord(station!.Value, component.ActiveKey!.Value, out GeneralCriminalRecord? record, stationRecordsComponent);
+
+        record!.Reason = secInfo!.Status == SecurityStatus.None ? string.Empty : secInfo.Reason;
+        record.Status = secInfo.Status;
+
+        _stationRecordsSystem.Synchronize(station.Value);
+        UpdateUserInterface(uid, component);
+    }
+
+    private void OnStatusSelected(EntityUid uid, GeneralCriminalRecordConsoleComponent component,
+        CriminalStatusOptionButtonSelected msg)
+    {
+        if (msg.Session.AttachedEntity is not {Valid: true} mob) return;
+        if (!CanUse(mob, uid))
+        {
+            _popupSystem.PopupEntity(Loc.GetString("general-criminal-record-permission-denied"), uid, msg.Session);
+            return;
+        }
+
+        TryComp<SecurityInfoComponent>(msg.Session.AttachedEntity, out var secInfo);
+
+        if (msg.Status == secInfo!.Status)
+            return;
+
+        if (msg.Reason != string.Empty && msg.Session.AttachedEntity != null)
+        {
+            secInfo.Status = secInfo.Status == SecurityStatus.None ? SecurityStatus.Wanted : SecurityStatus.None;
```

Comment:
this will actually set the security info of the mob that sent that message

in general, i think you should avoid doing something like that like we discussed, where setting a status on a record then attempts to set the status on an entity - again, record identification occurs through an ID card or whatever holds the key for a station record, and does not necessarily have to identify/be tied to a specific mob